### PR TITLE
Update docs for setting custom response code in exception handler

### PR DESCRIPTION
**Pull request by jameshalsall**
Docs update for the changes applied in https://github.com/symfony/symfony/pull/19822

### CODE REVIEW

File: reference/events.rst [hunk 1]
```diff
@@ -243,17 +243,20 @@ and set a new ``Exception`` object, or do nothing::
...
-        $response = new Response(
-            'Error',
-            404, // this status code will be ignored
-            array(
-                'X-Status-Code' => 200 // this status code will actually be sent to the client
-            )
-        );
+    response alone won't work. If you want to overwrite the status code
+    (which you should not without a good reason), call
+    ``GetResponseForExceptionEvent::setAllowSuccessfulResponse(true)`` and
```
Comment (fabpot) [inline reference/events.rst]: should be `setAllowCustomResponseCode` as the method has been renamed.

**Comment by xabbuh** [comment]
The code PR was merged.

@jameshalsall Thank you for the contribution. Can you fix the conflicts here?

**Comment by jameshalsall** [comment]
@xabbuh I will sort this out this week :)

**Comment by HeahDude** [comment]
@jameshalsall Will you have time to finish this PR or do you want we to take care of it? Thanks!

**Comment by javiereguiluz** [comment]
Thanks James.

**Pull request closed without merging**